ir/src/error.rs

```diff
@@ -75,4 +76,30 @@ impl SemanticError {
             matrix_col_len
         ))
     }
+
+    pub(super) fn invalid_list_folding(
+        lf_value_type: &air_script_core::ListFoldingValueType,
+        symbol_type: &IdentifierType,
+    ) -> SemanticError {
+        SemanticError::InvalidListFolding(format!(
+            "Symbol type {} is not supported for list folding value type {:?}",
+            symbol_type, lf_value_type
+        ))
+    }
+
+    pub(super) fn list_folding_empty_list(
+        lf_value_type: &air_script_core::ListFoldingValueType,
+    ) -> SemanticError {
+        SemanticError::InvalidListFolding(format!(
+            "List folding value cannot be an empty list. {:?} represents an empty list.",
+            lf_value_type
+        ))
+    }
+
+    pub(super) fn single_trace_column_allowed(ident: &str) -> SemanticError {
+        SemanticError::InvalidUsage(format!(
+            "{} should be a single trace column and not a group.",
+            ident
+        ))
+    }
```

Comment:
nit: can you update all of these for the new formatting?

e.g.
```
"Symbol type {symbol_type} is not supported for list folding value type {lf_value_type:?}"
```

Also, maybe it would be better to implement `Display` for the listfolding value type than to use debug printing here - but that could be a separate PR so that we can get this merged

Reply:
I haven't implemented the Display trait yet. Will open a separate PR.